unlock-app/src/hooks/useMultipleRecipient.ts

```diff
@@ -0,0 +1,198 @@
+import { useState, useEffect, useContext } from 'react'
+import { ToastHelper } from '../components/helpers/toast.helper'
+import { PaywallConfig } from '../unlockTypes'
+import { ConfigContext } from '../utils/withConfig'
+import { getAddressForName } from './useEns'
+
+const MAX_RETRY_COUNT = 5
+interface User {
+  lockAddress: string
+  userAddress: string
+}
+export interface RecipientItem {
+  userAddress: string
+  resolvedAddress: string
+  valid: boolean
+  index: number
+  metadata?: { [key: string]: any }
+  keyId?: string
+}
+
+interface RecipientPayload {
+  users: {
+    userAddress: string
+    metadata: { [key: string]: any }
+    lockAddress: string
+  }[]
+}
+
+export const useMultipleRecipient = (
+  paywallConfig?: PaywallConfig,
+  lockAddress?: string
+) => {
+  const [hasMultipleRecipients, setHasMultipleRecipients] = useState(false)
+  const [recipients, setRecipients] = useState(new Set<RecipientItem>())
+  const [loading, setLoading] = useState(false)
+  const config: any = useContext(ConfigContext)
+  const [retryCount, setRetryCount] = useState(0)
+  const [ignoreRecipients, setIgnoreRecipients] = useState<User[]>([])
+  const { maxRecipients = 1, locks } = paywallConfig ?? {}
+  const lockByAddress = lockAddress ? locks?.[lockAddress] : {}
+  const normalizeRecipients = () => {
+    if (!lockAddress) return
+
+    /*
+      we need to exclude user metadata that already already exists,
+      by creating a new array without items in "ignoreRecipients"
+     */
+    const listWithoutExcluded = recipientsList().filter(
+      ({ resolvedAddress }) => {
+        const exludedAddressList = (ignoreRecipients ?? []).map(
+          ({ userAddress }) => userAddress
+        )
+        const isAddressExcluded =
+          (exludedAddressList?.length > 0 &&
+            exludedAddressList.includes(resolvedAddress)) ||
+          exludedAddressList.length === 0
+        return isAddressExcluded
+      }
+    )
+
+    const payload: RecipientPayload = {
+      users: listWithoutExcluded.map(({ resolvedAddress, metadata = {} }) => {
+        return {
+          userAddress: resolvedAddress,
+          metadata,
+          lockAddress,
+        }
+      }),
+    }
+    return payload
+  }
+
+  const clear = () => {
+    setRecipients(new Set([]))
+  }
+
+  useEffect(() => {
+    const activeMultiple = +maxRecipients > 1
+    setHasMultipleRecipients(activeMultiple)
+  }, [])
+
+  const recipientsList = (): RecipientItem[] => {
+    return Array.from(recipients)
+  }
+
+  const canAddUser = () => {
+    return recipients.size < maxRecipients
+  }
+
+  const getAddressAndValidation = async (recipient: string) => {
+    const address = await getAddressForName(recipient)
+    return {
+      valid: address?.length > 0 ?? false,
```

Comment:
You can remove the extra `?? false` part safely here, I believe. The result would be the same.